src/settings.ts

```diff
@@ -13,36 +13,59 @@ export class Settings {
                 return;
             }
             const updatedConfig = workspace.getConfiguration("java.dependency");
-            if (updatedConfig.showOutline !== this._depdendencyConfig.showOutline
-                || updatedConfig.packagePresentation !== this._depdendencyConfig.packagePresentation) {
+            if (updatedConfig.showOutline !== this._dependencyConfig.showOutline
+                || updatedConfig.packagePresentation !== this._dependencyConfig.packagePresentation
+                || (updatedConfig.syncWithFolderExplorer !== this._dependencyConfig.syncWithFolderExplorer
+                    && updatedConfig.syncWithFolderExplorer)) {
+                this._dependencyConfig = updatedConfig;
```

Comment:
` this._dependencyConfig = updatedConfig;` duplicate during if and else

Reply:
We will call refresh when we found configuration changed,  and some functions triggered by refresh command will consult configurations, so i call  `this._dependencyConfig = updatedConfig` before execution.